### PR TITLE
Add simplified SWU method

### PR DESCRIPTION
**Pull request by davidnevadoc**
Add simplified SWU method for hashing to curves where `a*b !=0`.

PR changes:
 - Add SSWU method from https://datatracker.ietf.org/doc/html/rfc9380#name-simplified-shallue-van-de-w
 - Add SSWU's Z constant for secp256r1 https://datatracker.ietf.org/doc/html/rfc9380#sswu-z-code
 - Add hash to curve for secp256r1
 - Replace broken link comment for `svdw_hash_to_curve`

### CODE REVIEW

File: src/hash_to_curve.rs [hunk 1]
```diff
@@ -83,6 +83,94 @@ fn hash_to_field<F: FromUniformBytes<64>>(
...
+    domain_prefix: &'a str,
+    z: C::Base,
+) -> Box<dyn Fn(&[u8]) -> C + 'a>
+where
+    C: CurveExt,
+    C::Base: FromUniformBytes<64>,
+{
+    Box::new(move |message| {
+        let mut us = [C::Base::ZERO; 2];
+        hash_to_field("SVDW", curve_id, domain_prefix, message, &mut us);
```
Comment (han0110) [inline src/hash_to_curve.rs:164]: It seems the `MAP_ID` for Simplified SWU is `SSWU`

```suggestion
        hash_to_field("SSWU", curve_id, domain_prefix, message, &mut us);
```

File: src/secp256r1/curve.rs [hunk 2]
```diff
@@ -75,77 +76,97 @@ new_curve_impl!(
...
-    crate::tests::curve::curve_tests::<Secp256r1>();
+impl Secp256r1 {
+    // Optimal Z with: <https://datatracker.ietf.org/doc/html/rfc9380#sswu-z-code>
+    // 0xffffffff00000001000000000000000000000000fffffffffffffffffffffff5
+    const SSVDW_Z: Fp = Fp::from_raw([
+        0xfffffffffffffff5,
+        0x00000000ffffffff,
+        0x0000000000000000,
+        0xffffffff00000001,
+    ]);
```
Comment (han0110) [inline src/secp256r1/curve.rs:90]: Checked same as the suite `Z = -10` (https://www.rfc-editor.org/rfc/rfc9380.html#section-8.2)

**Review by han0110 — APPROVED** [review]
Overall LGTM! Just a small question.

File: src/hash_to_curve.rs (hunk 1 above)
Comment (davidnevadoc) [reply src/hash_to_curve.rs:164]: Good catch!

**Pull request merged**